pkg/virt-handler/migration-target.go

```diff
@@ -711,6 +718,13 @@ func (c *MigrationTargetController) processVMI(vmi *v1.VirtualMachineInstance) (
 
 	options := virtualMachineOptions(nil, 0, nil, c.capabilities, c.clusterConfig)
 	options.InterfaceDomainAttachment = domainspec.DomainAttachmentByInterfaceName(vmi.Spec.Domain.Devices.Interfaces, c.clusterConfig.GetNetworkBindings())
+
+	if c.clusterConfig.PasstIPStackMigrationEnabled() {
+		if err := c.passtRepairHandler.HandleMigrationTarget(vmi, c.passtSocketDirOnHostMigrationTarget); err != nil {
```

Comment:
Using a function pointer seems redundant here.
A simple `c.passtRepairHandler.HandleMigrationTarget(vmi, c.passtSocketDirOnHostMigrationTarget(vmi))` should work. For testing, you can just stub the `podIsolationDetector` to return whatever you need.

Same for the source.

Reply:
We have tried two options:
1. passt.RepairManager with isolationDetactor - makes pkg/network/passt depend on pkg/virt-handler/isolation
2. Calling `c.passtSocketDirOnHostMigrationTarget(vmi)` directly as you've suggested.
```go
	if c.clusterConfig.PasstIPStackMigrationEnabled() {
                passtPath, err := c.passtSocketDirOnHostMigrationTarget(vmi)
                if err == nil {
		    if err := c.passtRepairHandler.HandleMigrationTarget(vmi, passtPath); err != nil {
    			log.Log.Object(vmi).Warningf("failed to call passt-repair for migration target, %v", err)
   		    }
		}
	}
```

I think its not worth calling a relatively expensive function if for the majority of cases it will fail / its value will be discarded.

Reply:
Why not going with the 2nd approach, and also, if 
```
passtPath, err := c.passtSocketDirOnHostMigrationTarget(vmi)
```
returns "NotExists" error than there is nothing to do for the passt repair logic. The socket file is always there? or its created on demand?